### PR TITLE
R doc 1787 http config

### PR DESCRIPTION
**Pull request by morhilai**
HTTP Configuration options Http.Protocols and Http.UseLibuv
Unrelated code sample fix in HowToPerformFacetedSearch.cs

### CODE REVIEW

File: Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown [hunk 1]
```diff
@@ -104,8 +104,30 @@ Set the compression level to be used when compressing HTTP responses with Deflat
...
+- **Scope**: Server-wide only
+
+{PANEL/}
+
+{PANEL:Http.Protocols}
+
+Sets HTTP protocols that should be supported by the server.
+
+- **Type**: `enum HttpProtocols` ( `None = 0`, `Http1 = 1`, `Http2 = 2`, `Http1AndHttp2 = 3`)
+- **Default**: `DefaultValueSetInConstructor`
```
Comment (ppekrol) [inline Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown:130]: Describe what are the defaults. What is the logic there.

1: https://github.com/ravendb/ravendb/blob/5e42c31e7cd108eabd681cc32748c0c46c734f29/src/Raven.Server/Config/Categories/HttpConfiguration.cs#L16
2: https://github.com/ravendb/ravendb/blob/5e42c31e7cd108eabd681cc32748c0c46c734f29/src/Sparrow/Platform/PlatformDetails.cs#L40 (Windows 10, Windows Server 2016 or newer)

File: Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown [hunk 2]
```diff
@@ -104,8 +104,30 @@ Set the compression level to be used when compressing HTTP responses with Deflat
...
+- **Default**: `false`
+- **Scope**: Server-wide only
+
+{PANEL/}
+
+{PANEL:Http.Protocols}
+
+Sets HTTP protocols that should be supported by the server.
+
+- **Type**: `enum HttpProtocols` ( `None = 0`, `Http1 = 1`, `Http2 = 2`, `Http1AndHttp2 = 3`)
```
Comment (ppekrol) [inline Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown:129]: Avoid `= 0, = 1`. Not necessary. We prefer strings.

Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown:132]: Fixed by writing:

> By default, the HTTP protocol is set by the constructor of `HttpConfiguration` (that's what 
is meant by the value `DefaultValueSetInConstructor`).  
The HTTP protocol is chosen according to the value of `PlatformDetails.CanUseHttp2`, which is 
`true` if the platform running RavenD is Windows 10 or higher, or POSIX.

File: Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown [hunk 3]
```diff
@@ -126,8 +128,13 @@ Switches Kestrel to use Libuv.
...
 Sets HTTP protocols that should be supported by the server.
 
-- **Type**: `enum HttpProtocols` ( `None = 0`, `Http1 = 1`, `Http2 = 2`, `Http1AndHttp2 = 3`)
+- **Type**: `enum HttpProtocols` ( `None`, `Http1`, `Http2`, `Http1AndHttp2`)
 - **Default**: `DefaultValueSetInConstructor`
 - **Scope**: Server-wide only
 
+By default, the HTTP protocol is set by the constructor of `HttpConfiguration` (that's what 
+is meant by the value `DefaultValueSetInConstructor`).  
+The HTTP protocol is chosen according to the value of `PlatformDetails.CanUseHttp2`, which is
```
Comment (ppekrol) [inline Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown:137]: a) chosen to what value?
b) that is an implementation detail: 'PlatformDetails.CanUseHttp2'

Write sth like 'By default RavenDB supports HTTP1.1 on all Operating Systems. For Windows 10 (or newer), Windows Server 2016 (or newer) or on POSIX the support for HTTP2 is also turned on by default'

Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/configuration/http-configuration.markdown]: Done

**Pull request merged**